lib/merchant_repository.rb

```diff
@@ -61,6 +61,10 @@ def find_total_revenue_by_merchant_id(id)
     parent.find_total_revenue_by_merchant_id(id)
   end
 
+  def revenue(date=nil)
+    parent.find_total_revenue_by_merchant_id(id=nil, date)
```

Comment:
Useless assignment to variable - `id`.<br>Surrounding space missing for operator '='.